azurerm_cdn_endpoint/main.tf

```diff
@@ -23,6 +23,34 @@ resource "azurerm_cdn_endpoint" "cdn_endpoint" {
     host_name = var.origin_host_name
   }
 
+  dynamic "global_delivery_rule" {
+    iterator = l
+    for_each = var.global_delivery_rule_cache_expiration_action
+    content {
+      cache_expiration_action {
+        behavior = l.value.behavior
+        duration = l.value.duration
+      }
+    }
+  }
+
+  dynamic "delivery_rule" {
+    iterator = l
+    for_each = var.delivery_rule_url_path_condition_cache_expiration_action
```

Comment:
If you change the order or add something not at the end of the list all the rules appears as changed.
You can use a map as input object with "name" as key or create a map here using the same method used here:
https://github.com/pagopa/io-infrastructure-modules-new/blob/master/azurerm_monitor_autoscale_setting/main.tf#L18

Reply:
@francescopersico Mmm, ok. Following the method you suggested, I modified the code to take into account the `order` argument that should be more significant than the name. In this way:
- if I only change the position in the input list of the existing rules nothing appears as changed
- if I add a new rule in any position in the input list with `order` value greater than the other existing rules, only the new rule appears as to be created 
- if two different rules have the same `order` (key) regardless the position in the list an error is produced